crates/parcel_config/src/parcel_rc_config.rs

```diff
@@ -0,0 +1,781 @@
+use std::path::Path;
+use std::path::PathBuf;
+
+use package_manager::PackageManager;
+use parcel_filesystem::FileSystem;
+use pathdiff::diff_paths;
+
+use super::config_error::ConfigError;
+use super::parcel_config::ParcelConfig;
+use super::parcel_config::PluginNode;
+use super::parcel_rc::Extends;
+use super::parcel_rc::ParcelRcFile;
+use super::partial_parcel_config::PartialParcelConfig;
+
+#[derive(Default)]
+pub struct LoadConfigOptions<'a> {
+  pub additional_reporters: Vec<PluginNode>,
+  pub config: Option<&'a str>,
+  pub fallback_config: Option<&'a str>,
+}
+
+/// Loads and validates .parcel_rc config
+pub struct ParcelRcConfig<'a, T, U> {
+  fs: &'a T,
+  package_manager: &'a U,
+}
+
+impl<'a, T: FileSystem, U: PackageManager> ParcelRcConfig<'a, T, U> {
+  pub fn new(fs: &'a T, package_manager: &'a U) -> Self {
+    ParcelRcConfig {
+      fs,
+      package_manager,
+    }
+  }
+
+  fn find_config(&self, project_root: &Path, path: &PathBuf) -> Result<PathBuf, ConfigError> {
+    let from = path.parent().unwrap_or(path);
+
+    self
+      .fs
+      .find_ancestor_file(vec![String::from(".parcelrc")], from, project_root)
+      .ok_or(ConfigError::MissingParcelRc(PathBuf::from(from)))
+  }
+
+  fn resolve_from(&self, project_root: &PathBuf) -> PathBuf {
+    let cwd = self.fs.cwd().unwrap();
+    let relative = diff_paths(cwd.clone(), project_root);
+    let is_cwd_inside_project_root =
+      relative.is_some_and(|p| !p.starts_with("..") && !p.is_absolute());
+
+    let dir = if is_cwd_inside_project_root {
+      &cwd
+    } else {
+      project_root
+    };
+
+    dir.join("index")
+  }
+
+  fn load_config(&self, path: PathBuf) -> Result<(PartialParcelConfig, Vec<PathBuf>), ConfigError> {
```

Comment:
Would you mind documenting this module ; what it does, how it traverses directories and merges configuration

Reply:
Sure